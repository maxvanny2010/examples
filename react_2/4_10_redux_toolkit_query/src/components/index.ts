export * from './Cart';
export * from './Header';
export * from './OrderModal';
export * from './ParentComponent';
export * from './Product';
export * from './Total';
