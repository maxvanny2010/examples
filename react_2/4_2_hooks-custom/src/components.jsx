export { Toggle } from './Toggle.jsx';
export { Storage } from './Storage.jsx';
export { Debounce } from './Debounce.jsx';
export { Timeout } from './Timeout.jsx';
export { Array } from './Array.jsx';
export { Throttle } from './Throttle.jsx';
export { UpdateEffect } from './UpdateEffect.jsx';
