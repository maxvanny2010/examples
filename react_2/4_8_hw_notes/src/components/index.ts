export { AuthForm } from './AuthForm';
export { NotesList } from './NotesList';
export { NoteDetail } from './NoteDetail';
export { UserInfoBar } from './UserInfoBar';
export { NotesSearch } from './NotesSearch';
export { CodeBlockMarkdown } from './CodeBlockMarkdown';
export { DeleteConfirmDialog } from './DeleteConfirmDialog';
