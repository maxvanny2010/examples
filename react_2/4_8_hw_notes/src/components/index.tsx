export { AuthForm } from './AuthForm';
export { NotesList } from './NotesList';
export { NotesSearch } from './NotesSearch';
export { NoteDetail } from './NoteDetail';
export { DeleteConfirmDialog } from './DeleteConfirmDialog';
export { CodeBlockMarkdown } from './CodeBlockMarkdown';
export { UserInfoBar } from './UserInfoBar';
