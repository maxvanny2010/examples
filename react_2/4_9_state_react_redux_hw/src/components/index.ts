export * from './Breadcrumbs';
export * from './ContactCard';
export * from './Empty';
export * from './FilterForm';
export * from './GroupContactsCard';
export * from './Layout';
export * from './MainMenu';
export * from './FavoriteToggleButton';
export * from './ContactListItem';
export * from './FilterFormWrapper';
export * from './RowMemo';