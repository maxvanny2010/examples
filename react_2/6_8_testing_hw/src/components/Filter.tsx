type Props = {
	currentFilter: 'all' | 'active' | 'completed';
	onChange: (filter: 'all' | 'active' | 'completed') => void;
};

export const Filter = ({ currentFilter, onChange }: Props) => {
	return (
		<div style={{ marginTop: '1rem' }}>
			<button
				onClick={() => onChange('all')}
				disabled={currentFilter === 'all'}
			>
				Все
			</button>
			<button
				onClick={() => onChange('active')}
				disabled={currentFilter === 'active'}
			>
				Активные
			</button>
			<button
				onClick={() => onChange('completed')}
				disabled={currentFilter === 'completed'}
			>
				Выполненные
			</button>
		</div>
	);
};
