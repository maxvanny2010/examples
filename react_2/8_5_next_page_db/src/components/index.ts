export * from './ui/Alert';
export * from './ui/HomeButton';
export * from './ui/LoginLink';
export * from './ui/Layout';