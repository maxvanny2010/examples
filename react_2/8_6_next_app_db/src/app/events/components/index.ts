export * from './ButtonEventJoin';
export * from './EventList';
export * from './CardEvent';
export * from './StateLoading';
export * from './StateError';
export * from './StateEmpty';
export * from './ForbiddenCard';
export * from './UnauthorizedCard';
