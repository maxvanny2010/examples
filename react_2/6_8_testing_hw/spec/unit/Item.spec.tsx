import { Item } from 'src/components/Item';
import { render, screen } from '@testing-library/react';

describe('Элемент списка задач', () => {
	it.todo('название не должно быть больше 32 символов');
	it.todo('название не должно быть пустым');
	it.todo('нельзя удалять невыполненные задачи');

	it('не отображает полное название, если оно слишком длинное', () => {
		const longTitle = 'Это очень очень очень длинное название задачи, которое превышает 32 символа';
		render(<Item id="1"
					 header={longTitle}
					 done={false}
					 onDelete={() => {
					 }}
					 onToggle={() => {
					 }} />);
		expect(screen.getByText(/очень длинное название/i)).toBeInTheDocument();
	});
	it('удаление недоступно, если задача не завершена', () => {
		render(<Item id="1"
					 header="Тест"
					 done={false}
					 onDelete={() => {
					 }}
					 onToggle={() => {
					 }} />);
		const button = screen.getByRole('button');
		expect(button).toBeDisabled();
	});
	it('корректно отображает задачу (snapshot)', () => {
		const { container } = render(
			<Item id="1"
				  header="Снапшот"
				  done={false}
				  onDelete={() => {
				  }}
				  onToggle={() => {
				  }} />,
		);
		expect(container).toMatchSnapshot();
	});

});