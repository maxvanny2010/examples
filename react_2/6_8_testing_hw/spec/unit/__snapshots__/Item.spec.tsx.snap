// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Элемент списка задач корректно отображает задачу (snapshot) 1`] = `
<div>
  <li
    class="item-wrapper"
  >
    <input
      id="1"
      type="checkbox"
    />
    <label
      for="1"
    >
      Снапшот
    </label>
    <button
      class="button button-with-icon"
      data-alt="Удалить"
      disabled=""
    >
      <img
        alt="Удалить"
        src="[object Object]"
      />
    </button>
  </li>
</div>
`;
